vignettes/rhino-fluent.Rmd

```diff
@@ -0,0 +1,560 @@
+
+The style for the class `ms-depth-8` is added in the `app/style/main.scss` file. So the file now is modified to -
+```
+# app/style/main.scss
+
+.ms-depth-4{
+    display: flex; 
+    flex-wrap: wrap;
+    padding: 5px;
+}
+
+.ms-depth-8{
+    height: 500px; 
+    overflow: auto; 
+    width: 50%; 
+    display: inline-block;
+}
+```
+
+The Sass code limits the height of the table to 500 pixels, width to 50% of the screen size and allows scrolling through the data. Similar to the previous usecase, we would again have to repeat the Sass building process with `rhino` as the file has been updated.
+
+```
+# in R console
+rhino::build_sass()
+```
+
+So, this is how the UI should look now -
+
+```{r, echo=FALSE}
+knitr::include_graphics("images/rhino_fluent_gif_2.gif")
+```
+Note that we now we have space beside the table for additional UI component(s).
+
+## Render a Bar Chart with logic in a helper function
+
+We will now add a bar chart that plots the deal amounts closed by the sales reps. For this, we will code a helper function and keep it inside the `app/logic` folder.
+
+### Helper function in the logic folder of the `rhino` project
+
+* Go to the `app/logic` folder and create a new `.R` file. Lets name it `barchart.R`.
+* We shall use the `gglplot2` and `plotly` packages (which we installed in the beginning) for this purpose. 
+
+Here is the code -
+
+```
+# app/logic/barchart.R
+
+box::use(
+  ggplot2[ggplot, aes, geom_bar, labs, theme_light, theme, element_text],
+)
+
+#' @export
+generate_barchart <- function(filtered_data) {
+        ggplot(filtered_data, aes(x = rep_name, y = deal_amount, fill = rep_name)) +
+        geom_bar(stat = "identity") +
+        labs(x = "Sales Rep Name", y = "Deal Amount") +
+        theme_light() +
+        theme(axis.text.x = element_text(angle = 60, hjust = 1))
+}
+```
+
+The above code block loads the `ggplot2` package and imports several functions from it. It then defines a function called `generate_barchart()` that takes a `filtered_data` argument as input. The function uses the `ggplot()` function to create a bar chart with `x = rep_name` and `y = deal_amount`, where `rep_name` and `deal_amount` are columns in the `filtered_data` data frame. The `fill` aesthetic is also set to `rep_name`, which maps each bar to a unique color based on the `rep_name` variable. The function also adds labels to the x and y axes and applies the `theme_light()` theme to the plot.
+
+The `element_text()` function takes several arguments to customize the text appearance, including the `angle` argument to specify the rotation angle in degrees. Here, we set the angle to 60 degrees to rotate the labels slightly vertically, and we set the `hjust` argument to 1 to align the labels to the right of the tick marks.
+
+### Call the helper function in the desired `rhino` module
```

Comment:
The chart should have its own module. And this is the opportunity to also move the toggle to a separate one. Check the structure in the main Rhino tutorial.